My VA - Payment info cleanup

allisonlu opened the pull request:
## Summary
My VA updated the Benefit payments and debts section so that it is now two sections: 
- Outstanding debts and 
- Benefit payments

This PR removes any instances of the feature flag (this is now launched to 100% of users), and updates e2e tests. 


## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/47243

## Testing done
- all unit and e2e tests pass

## What areas of the site does it impact?
My VA

## Acceptance criteria
- [x] Toggle is removed in FE code and API
- [x] Conditionals are removed (all users are seeing only v2 of payment info - ie two separate debt and benefit sections)
- [x] Tests are updated

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ out of scope errors ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

allisonlu committed:
removed showPaymentAndDebtSection feature toggle

allisonlu committed:
committing a successful test before i work on the
rest

allisonlu committed:
updated e2e tests that were using the removed
showPaymentAndDebt feature toggle

allisonlu committed:
remove e2e tests for old section

adamwhitlock1 commented on src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2.cypress.spec.js:
```diff
@@ -42,38 +53,8 @@ describe('The My VA Dashboard - Outstanding Debts', () => {
-    });
-  });
-
   describe('when the feature is not hidden', () => {
```
This looks good, but maybe we should change up this describe to be more along the lines of 'when feature renders correctly' or something like that, since we aren't hiding it at all now via feature toggles.

adamwhitlock1 reviewed (commented):
Great job sifting through this logic and getting the feature toggle stuff pulled out.

I had a few thoughts mostly on the tests and some non-blocking considerations, but the only main/blocking concern I have is the removal of the check for `!showMPIConnectionError && !showNotInMPIError`

I think we have a couple e2e tests around these cases, maybe we could add something to those to check this logic... I'm still thinking about this and will mess around with this branch some more to see if I can reproduce these conditions.

allisonlu commented on src/applications/personalization/dashboard/components/Dashboard.jsx:
```diff
@@ -256,15 +253,7 @@ const Dashboard = ({
-                />
-              ) : null}
-              {showBenefitPaymentsAndDebtV2 ? (
+              {isLOA3 && shouldShowV2Dashboard ? (
```
The Benefit payments section should show for all LOA3 users, even if user doesn't have any recent payments, so I removed the check for payment history. It's also being [checked in the section component](https://github.com/department-of-veterans-affairs/vets-website/blob/e03827e198bceb4c1452fa6c1bd93d47b8450e54/src/applications/personalization/dashboard/components/benefit-payments-v2/BenefitPaymentsV2.jsx#L154).

But I'll remove the ternary 👍

allisonlu replied:
Oop, I take that back. We're using `canAccessPaymentHistory` to call `getAllPayments` from the payments action, and passing this array into `BenefitPaymentsV2`. I'll add this conditional back in.

allisonlu committed:
- using .findAllByTestId instad of .findByTestId
or .get
- updating some test names to be more specific
- removed "feature is hidden" tests

allisonlu commented on src/applications/personalization/dashboard/tests/e2e/outstanding-debts-v2-error-states.cypress.spec.js:
```diff
@@ -64,20 +64,23 @@ describe('My VA - Outstanding debts error-states', () => {
 
-      cy.findByTestId('outstanding-debts-error').should('exist');
+      // findByTestId for the commented out elements are flakey (for this test spec only...)
+      // cy.findByTestId('outstanding-debts-error').should('exist');
+      cy.get('[data-testid="outstanding-debts-error"]').should('exist');
       cy.findByTestId('debt-card-v2').should('not.exist');
       cy.findByTestId('copay-card-v2').should('exist');
       cy.findByTestId('copay-due-header-v2').should('exist');
       cy.findByTestId('manage-va-copays-link-v2').should('exist');
-      cy.findByTestId('learn-va-debt-link-v2').should('exist');
+      // cy.findByTestId('learn-va-debt-link-v2').should('exist');
+      cy.get('[data-testid="learn-va-debt-link-v2"]').should('exist');
```
It's kinda bizarre because it's only in that one test spec where `findByTestId` is whacky, and adding `@features` back in didn't help. The error I was getting in the terminal is "found multiple elements", so I replaced it with `findAllByTestId`, and all tests pass :shrug:

adamwhitlock1 commented on src/applications/personalization/dashboard/components/Dashboard.jsx:
```diff
@@ -344,11 +333,6 @@ const mapStateToProps = state => {
     !showNotInMPIError &&
     isLOA3 &&
     isVAPatient;
-  const showBenefitPaymentsAndDebt =
-    !showMPIConnectionError && !showNotInMPIError && isLOA3;
```
So I started poking a bit more at what the existing My VA looks in the event of an MPI error and kinda seeing what the UI state was.

I opened up cypress and the `mpi-connection-error.cypress.spec.js` file. 
I then added:
``` js
cy.intercept(
        '/v0/feature_toggles*',
        generateFeatureToggles({
          showMyVADashboardV2: true,
        }),
      );
```
to the beforeEach section and switched between this branch and main to see the diff

## MAIN:
![Screenshot 2023-06-02 at 9 47 03 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/ec49590a-9570-4c96-8e69-8df432765892)

## This Branch
![Screenshot 2023-06-02 at 9 45 35 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/b838420a-105d-4b61-b34b-ff832eb492a7)

So there are some differences, but there are at least error boxes for each debt/payment section. I'm not sure if this is what we want, or if some changes need to be made based on the designs in place, but wanted to put my findings in here for consideration.

allisonlu replied:
If there is an MPI connection, user should see a [full-page error](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/my-va/use-cases/system-cant-connect-to-MPI.md).

allisonlu committed:
- added back payment history state check for
BenefitPaymentsV2

adamwhitlock1 reviewed (approved):
Most recent changes and comments look good to me.

adamwhitlock1 commented on src/applications/personalization/dashboard/mocks/feature-toggles/index.js:
```diff
@@ -3,7 +3,6 @@ const generateFeatureToggles = (toggles = {}) => {
     myVaUseExperimental = true,
```
oh, this is getting flagged by eslint, so might just need some minor cleanup of formatting

allisonlu committed:
Merge branch 'main' into My-VA/allison/47243/payment-info-cleanup

allisonlu committed:
fix eslint issue
